server/db/actions/Dog.js

```diff
@@ -0,0 +1,59 @@
+import dbConnect from "../dbConnect";
+import Dog from "../models/Dog";
+import Log from "../models/Log";
+import User from "../models/User";
+
+/**
+ * Creates a new dog in the database
+ * @param {*} dogData Object that has been parsed by Zod for validity
+ * @returns string id of the dog if successfully saved, error otherwise
+ */
+export async function createDog(dogData) {
+  await dbConnect();
```

Comment:
I would wrap this in a try/catch with a custom error message that is more user friendly. Right now, if this function errors because we are unable to connect to the db, the mongoose error message is thrown (and we don't want the client seeing these dev-specific errors). 

Instead, we can do something like this:
```js
try {
    await dbConnect();
} catch (e) {
    throw new Error("Unable to create dog at this time, please try again");
}
```